src/features/device-state/state-get-utils.ts

```diff
@@ -65,10 +72,44 @@ export const getConfig = (
 	return config;
 };
 
-export const serviceInstallFromImage = (
+type ExpandedService = Pick<Service['Read'], 'id' | 'service_name'> & {
+	service_environment_variable: EnvVarList;
+	service_label: Array<{ label_name: string; value: string }>;
+};
+
+type ExpandedServiceInstall = {
+	id: ServiceInstall['Read']['id'];
+	service: ExpandedService[];
+	device_service_environment_variable: EnvVarList;
+};
+
+export function serviceInstallFromImage(
+	deviceOrFleet: { service_install: ExpandedServiceInstall[] },
+	image?: {
+		is_a_build_of__service: number | { __id: number };
+	},
+): ExpandedServiceInstall | undefined;
+
+export function serviceInstallFromImage(
+	deviceOrFleet: { service: ExpandedService[] },
+	image?: {
+		is_a_build_of__service: number | { __id: number };
+	},
+): ExpandedService | undefined;
+
+export function serviceInstallFromImage(
```

Comment:
Could you remove the empty lines between overloads here please, with the empty line it implies to me they are separate functions rather than a single (overloaded) function
```suggestion
export function serviceInstallFromImage(
	deviceOrFleet: { service_install: ExpandedServiceInstall[] },
	image?: {
		is_a_build_of__service: number | { __id: number };
	},
): ExpandedServiceInstall | undefined;
export function serviceInstallFromImage(
	deviceOrFleet: { service: ExpandedService[] },
	image?: {
		is_a_build_of__service: number | { __id: number };
	},
): ExpandedService | undefined;
export function serviceInstallFromImage(
```